corehq/apps/reports/analytics/esaccessors.py

```diff
@@ -574,43 +575,28 @@ def get_username_in_last_form_user_id_submitted(domain, user_id):
         return user_submissions[0]['form']['meta'].get('username', None)
 
 
-def get_wrapped_ledger_values(domain, case_ids, section_id, entry_ids=None, pagination=None):
-    # todo: figure out why this causes circular import
-    from corehq.apps.reports.commtrack.util import StockLedgerValueWrapper
-    query = (LedgerES()
-             .domain(domain)
-             .section(section_id)
-             .case(case_ids))
-    if pagination:
-        query = query.size(pagination.count).start(pagination.start)
-    if entry_ids:
-        query = query.entry(entry_ids)
-
-    return [StockLedgerValueWrapper.wrap(row) for row in query.run().hits]
+def get_wrapped_ledger_values(domain, case_ids, section_id, entry_ids=None):
+    return LedgerAccessors(domain).get_ledger_values_for_cases(case_ids, section_id, entry_ids)
 
 
 def products_with_ledgers(domain, case_ids, section_id, entry_ids=None):
-    # returns entry ids/product ids that have associated ledgers
-    query = LedgerES().domain(domain).section(section_id).case(case_ids)
-    if entry_ids:
-        query = query.entry(entry_ids)
-    return set(query.values_list('entry_id', flat=True))
+    return {
+        ledger.entry_id
+        for ledger in get_wrapped_ledger_values(domain, case_ids, section_id, entry_ids)
+    }
 
 
 def get_aggregated_ledger_values(domain, case_ids, section_id, entry_ids=None):
-    # todo: figure out why this causes circular import
-    query = LedgerES().domain(domain).section(section_id).case(case_ids)
-    if entry_ids:
-        query = query.entry(entry_ids)
-
-    terms = [
-        AggregationTerm('entry_id', 'entry_id'),
+    ledgers = get_wrapped_ledger_values(domain, case_ids, section_id, entry_ids)
+    ret = defaultdict(lambda: 0)
+    for ledger in ledgers:
+        ret[ledger.entry_id] += ledger.balance
```

Comment:
I'd like for this to be done with a single query (or at least one per db), but probably not worth it at this time